invenio_rdm_records/github/release.py

```diff
@@ -0,0 +1,315 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+"""Github release API implementation."""
+
+import requests
+import yaml
+from flask import current_app
+from invenio_github.api import GitHubRelease
+from invenio_github.models import ReleaseStatus
+from marshmallow import ValidationError
+from mistune import markdown
+from nameparser import HumanName
+
+from invenio_access.permissions import system_identity
+from invenio_db import db
+from invenio_pidstore.models import PersistentIdentifier
+from invenio_rdm_records.github.utils import get_extra_metadata, load_citation_metadata
+from invenio_rdm_records.proxies import current_rdm_records_service
+from invenio_records_resources.services.uow import UnitOfWork
+
+
+class RDMGithubRelease(GitHubRelease):
```

Comment:
This class seems to:

- Extract information from a release and create a dictionary of data via underscored methods (see comment below)
- Do record resolution manually, e.g. calls to `PersistentIdentifiers` and implement a pseudo resolver. Can we use existing resolvers/api classes?
- Process the release (missing validation?)
- Publish the release via rdm-records services

It seems to be a bit cluttered between an api class and a "service" (does not have to be a full fledge one, but split what's an object with information and data layer and what's management of it with service calls).

Reply:
Point-by-point replies:

1 - totally agree :)
2 - I ended up using `PersistentIdentifier.get_by_object_uuid` because we associate the record id (UUID) to the release as a weak reference to the record. Therefore, when we have a release with an associated record UUID, we can't use the PID resolver as it expects a `pid_value` and not `object_uuid`.  I did not find any resolver for such a case but maybe I missed it
3 - see the reply below. This method is part of the public interface of `invenio_github.api.GithubRelease` and, in this case, simply calls the record publishing part. Which validation do you refer to?